cpp/json_schema_converter.cc

```diff
@@ -698,10 +842,19 @@ std::string JSONSchemaConverter::VisitString(
       {
           "minLength",
           "maxLength",
-          "pattern",
           "format",
```

Comment:
Could you add support for `format` keyword as well? We can support `ipv4`, `ipv6`, `date-time`, `date`, `time`, `email` now. Check out https://github.com/mlc-ai/xgrammar/blob/main/tests/python/test_regex_converter.py, and https://json-schema.org/understanding-json-schema/reference/string.